butterknife-compiler/src/test/java/butterknife/UnbinderTest.java

```diff
@@ -190,68 +209,312 @@
+                "  }",
+                "  public static class Unbinder<T extends TestOne> extends Test$$ViewBinder.Unbinder<T> {",
+                "    View view1;",
+                "    Unbinder(T target) {",
+                "      super(target);",
+                "    }",
+                "    @Override protected void unbind(T target) {",
+                "      super.unbind(target);",
+                "      view1.setOnClickListener(null);",
+                "      target.unbinder2 = null;",
+                "    }",
+                "  }",
+                "}"
+            ));
+
+    assertAbout(javaSource()).that(source)
+        .processedWith(new ButterKnifeProcessor())
+        .compilesWithoutError()
+        .and()
+        .generatesSources(expectedSource1, expectedSource2);
+  }
+
+  @Test public void childUsesOwnUnbinder() {
+    JavaFileObject source = JavaFileObjects.forSourceString("test.Test",
+        Joiner.on('\n')
+            .join(
+                "package test;",
+                "import android.support.v4.app.Fragment;",
+                "import butterknife.ButterKnife;",
+                "import butterknife.OnClick;",
+                "import butterknife.Unbinder;",
+                "public class Test extends Fragment {",
+                "  @Unbinder ButterKnife.Unbinder unbinder;",
+                "  @OnClick(1) void doStuff1() { }",
+                "}",
+                "class TestOne extends Test {",
+                "  @OnClick(1) void doStuff2() { }",
+                "}"
+            ));
+
+    JavaFileObject expectedSource1 = JavaFileObjects.forSourceString("test/Test$$ViewBinder",
+        Joiner.on('\n')
+            .join(
+                "package test;",
+                "import android.view.View;",
+                "import butterknife.internal.DebouncingOnClickListener;",
+                "import butterknife.internal.Finder;",
+                "import butterknife.internal.ViewBinder;",
+                "import java.lang.IllegalStateException;",
+                "import java.lang.Object;",
+                "import java.lang.Override;",
+                "public class Test$$ViewBinder<T extends Test> implements ViewBinder<T> {",
+                "  @Override public void bind(final Finder finder, final T target, Object source) {",
+                "    Unbinder unbinder = (Unbinder) createUnbinder(target);",
+                "    View view;",
+                "    view = finder.findRequiredView(source, 1, \"method 'doStuff1'\");",
+                "    unbinder.view1 = view;",
+                "    view.setOnClickListener(new DebouncingOnClickListener() {",
+                "      @Override public void doClick(View p0) {",
+                "        target.doStuff1();",
+                "      }",
+                "    });",
+                "    target.unbinder = unbinder;",
+                "  }",
+                "  protected butterknife.ButterKnife.Unbinder<T> createUnbinder(T target) {",
```

Comment:
This could be `<U extends Unbinder<T>> U createUnbinder(T target)` to avoid the cast in calling code